### PR TITLE
Add the fix of the parallelepiped colorization made by Timo in Deal.II to the compat.h file in aspect.

### PR DESCRIPTION
**Pull request by MFraters**
Add the fix of the parallelepiped colorization made by Timo in Deal.II to the compat.h file in aspect. This is needed to continue with the ellipsoidal chunk fixes pull request.

### CODE REVIEW

File: include/aspect/compat.h [hunk 1]
```diff
@@ -22,6 +22,209 @@
...
+#if !DEAL_II_VERSION_GTE(8,4,0)
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_reordering.h>
+namespace dealii
+{
+  namespace GridGenerator
+  {
+    template<int dim>
+    void
+    parallelepiped (Triangulation<dim>  &tria,
```
Comment (tjhei) [inline include/aspect/compat.h]: Do you need parallelepiped or only subdivided_parallelepiped?

Comment (tjhei) [inline include/aspect/compat.h]: Also, which version of subdivided_parallelepiped do you need? If it is the second one, remove the first one here.

**Comment by tjhei** [comment]
(looks good otherwise)

File: include/aspect/compat.h (hunk 1 above)
Comment (MFraters) [inline include/aspect/compat.h]: I personally only need the subdivided version where the subdivisions are given by a vector. Remove the other ones?

**Comment by tjhei** [comment]
Yes.

File: include/aspect/compat.h [hunk 2]
```diff
@@ -22,6 +22,174 @@
...
+        {
+          typename Triangulation<dim>::active_cell_iterator
+          cell = tria.begin_active(),
+          endc = tria.end();
+          for (; cell!=endc; ++cell)
+            {
+              for (unsigned int face=0; face<GeometryInfo<dim>::faces_per_cell; ++face)
+                {
+                  if (cell->face(face)->at_boundary())
+                    cell->face(face)->set_boundary_id(face);
```
Comment (tjhei) [inline include/aspect/compat.h]: You need an ifdef to use ```set_boundary_indicator()``` for older deal.II versions:
```
#if DEAL_II_VERSION_GTE(8,3,0)
              cell->face(f)->set_boundary_id (f);
#else
              cell->face(f)->set_boundary_indicator (f);
#endif

```

**Comment by tjhei** [comment]
Thanks. Have you checked this is working? I think it would make more sense to include this change with your code that actually uses it.

**Comment by MFraters** [comment]
Yes, I checked it. But I can also include it into the other pull request. It makes indeed more sense now it only includes the one needed by the ellipsoid pull request. 


Van: Timo Heister

Verstuurd: zaterdag 17 oktober 13:42

Onderwerp: Re: [aspect] Add the fix of the parallelepiped colorization made by Timo in Deal.II to the compat.h file in aspect. (#645)

Aan: geodynamics/aspect

Cc: MFraters



Thanks. Have you checked this is working? I think it would make more sense to include this change with your code that actually uses it.

**Pull request closed without merging**